jgrapht-core/src/main/java/org/jgrapht/alg/CyclicTransitiveReduction.java

```diff
@@ -0,0 +1,297 @@
+/*
+ * (C) Copyright 2018-2020, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg;
```

Comment:
I leave this internal stuff to the maintainers. I have no desire to learn all your internal rules.

Reply:
Just curious:  when you visit someone's house, if they ask you to hang up your coat in a particular closet, do you instead just throw it on the couch?

Reply:
No, but this is not the same situation. I am not visiting your house because you won't even let me enter the room. I would be much more inclined to hang my coat where you specify if I felt more welcome. Even then, as a host you could take my coat and hang it where and how you please.

Why are we even discussing this? This kind of refactoring can be done by a maintainer in any IDE in 10 seconds. I put my algorithms in the packages where similar ones already exist. If you want to refactor, do it after accepting the contribution. I know I am repeating myself, but you are micro-managing me, trying to pull my strings like I was a puppet. The debate takes much longer than you doing what you prefer after accepting the contribution.

I have reacted open-mindedly to many of your suggestions and implemented them. But this kind of stuff really is irrelevant to the functionality, it is purely a structural attribute. Maintainers should deal with that. Contributing to a project should be easy and not a big hurdle scaring off potential contributors because they surely won't try a second time. Don't make this feel like rocket science.

Reply:
Who are these maintainers of which you speak?

Reply:
Sorry for not explaining what I thought was an obvious term:

By **maintainer** I mean someone who has direct commit rights to this repository and/or by her code review is part of the decision if a contribution is to be accepted into the product or not. Maintainers may have other responsibilities, such as build and publish releases. By contrast, a **contributor** is someone like me, outside of the core team without any special rights or privileges, who contributes to the project by means of PRs, raising and commenting on issues, updating wiki pages etc.

Both maintainers and contributors can be rather inactive or regularly active, my differentiation is unrelated to the level of activity or number or frequency of contribution.

Reply:
We don't have any maintainers.  We have committers.  Committers review code and help contributors (via comments) to get it into suitable shape for being merged.  That's our process, it's well documented, and it has worked quite effectively for us--take a look back over the history of all the other contributors whose work got merged.  Since that process doesn't work for you, there's no more to be said.

But if you do find a maintainer with the necessary rights, please feel free to dictate to her whatever you think best.

Reply:
> We don't have any maintainers. We have committers.

You call them committers, I call them maintainers because they do more than just commit. They effectively maintain the project. 

>  if you do find a maintainer with the necessary rights, please feel free to dictate to her whatever you think best

Who is dictating? You put me in a position to either do exactly as **you** dictate or refuse my contribution due to minor structural issues we disagree upon and which you could easily fix by yourself after accepting the code. That is what a maintainer does where I come from. I am sorry to learn that this kind of thing is beneath you. If you want to lead, don't tell other people to run up a hill because you say so. Take the flag and walk first.

_"And we lead not by the example of our power, but by the power of our example."_
Joe Biden